docs/types/reference-types.rst

```diff
@@ -468,6 +468,117 @@ Array Members
         }
     }
 
+.. index:: ! array;dangling storage references
+
+Dangling References to Storage Array Elements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When working with storage arrays, you need to take care to avoid dangling references.
```

Comment:
What about memory arrays? There's really nothing outside of inline assembly that can give you a dangling reference in memory?